src/main/java/com/commercetools/helper/mapper/impl/PaymentMapperImpl.java

```diff
@@ -0,0 +1,209 @@
+package com.commercetools.helper.mapper.impl;
+
+import com.commercetools.helper.formatter.PaypalPlusFormatter;
+import com.commercetools.helper.mapper.PaymentMapper;
+import com.commercetools.model.CtpPaymentWithCart;
+import com.paypal.api.payments.*;
+import io.sphere.sdk.cartdiscounts.DiscountedLineItemPriceForQuantity;
+import io.sphere.sdk.carts.CustomLineItem;
+import io.sphere.sdk.carts.LineItem;
+import io.sphere.sdk.models.LocalizedString;
+import org.javamoney.moneta.Money;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.money.MonetaryAmount;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Stream;
+
+import static com.commercetools.payment.constants.paypalPlus.PaypalPlusPaymentIntent.SALE;
+import static com.commercetools.util.MoneyUtil.getActualShippingCost;
+import static com.commercetools.util.MoneyUtil.getActualTax;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Stream.concat;
+
+@Component
+public class PaymentMapperImpl implements PaymentMapper {
+
+    private final PaypalPlusFormatter paypalPlusFormatter;
+
+    @Autowired
+    public PaymentMapperImpl(@Nonnull PaypalPlusFormatter paypalPlusFormatter) {
+        this.paypalPlusFormatter = paypalPlusFormatter;
+    }
+
+    @Override
+    @Nullable
+    public Payment ctpPaymentToPaypalPlus(@Nullable CtpPaymentWithCart paymentWithCartLike) {
+        if (paymentWithCartLike == null) {
+            return null;
+        }
+
+        final Payment mappedPayment = new Payment();
+
+        mappedPayment.setCart(paymentWithCartLike.getCart().getId());
+        mappedPayment.setIntent(SALE);
+        mappedPayment.setPayer(getPayer(paymentWithCartLike));
+        mappedPayment.setTransactions(getTransactions(paymentWithCartLike));
+        mappedPayment.setRedirectUrls(getRedirectUrls(paymentWithCartLike));
+
+        return mappedPayment;
+    }
+
+    @Nonnull
+    protected Payer getPayer(@Nonnull CtpPaymentWithCart paymentWithCartLike) {
```

Comment:
obviously - to override them if needed :)
this class could be extended in the future if some payments could be mapped differently based on some conditions (payment type and so on)